### PR TITLE
ci: make dev tools target fail when any of the dependencies fail

### PR DESCRIPTION
**Pull request by slonka**
make dev tools target fail when any of the dependencies install fails so that it will not get cached

Signed-off-by: slonka <slonka@users.noreply.github.com>

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: mk/dependencies/kubebuilder.sh [hunk 1]
```diff
@@ -1,10 +1,20 @@
 #!/bin/bash
 
+set -e
+
+if [ "$ARCH" == "arm64" ]; then
+  if [ "$OS" == "darwin" ]; then
+    # there is no arm64 build for darwin so use amd64 via rosetta
+    ARCH="amd64"
```
Comment (slonka) [inline mk/dependencies/kubebuilder.sh:8]: this never worked on m1 - https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v2.3.2

File: mk/dev.mk [hunk 2]
```diff
@@ -46,12 +46,7 @@ TOOLS_DEPS_LOCK_FILE=mk/dependencies/deps.lock
 # it's important that everything lands in $(CI_TOOLS_DIR) to be able to cache this folder in CI and speed up the build.
 .PHONY: dev/tools
 dev/tools: ## Bootstrap: Install all development tools
-	@mkdir -p $(CI_TOOLS_BIN_DIR) $(CI_TOOLS_DIR)/protos
-	# Also compute a hash to use for caching
-	FILES=`find $(TOOLS_DEPS_DIRS) -name '*.sh' | sort`; \
-		for i in $${FILES}; do OS=$(GOOS) ARCH=$(GOARCH) $$i ${CI_TOOLS_DIR}; done; \
```
Comment (michaelbeaumont) [inline mk/dev.mk:52]: the original issue here is that `make` doesn't break when an iteration fails. so these needed a `|| exit 1`. maybe we need to consistently set `.SHELLFLAGS` to add `-ec`.

**Comment by lahabana** [comment]
@slonka should this be merged?

**Comment by slonka** [comment]
> should this be merged?

Yes, I just need to figure out why it failed. Will be ready tomorrow. Also, maybe @michaelbeaumont approach is better, will test it out.

**Pull request merged**